crates/holochain/src/conductor/conductor.rs

```diff
@@ -1342,22 +1342,17 @@ mod app_impls {
         pub(crate) async fn install_app_legacy(
             self: Arc<Self>,
             installed_app_id: InstalledAppId,
-            cell_data: Vec<(InstalledCell, Option<MembraneProof>)>,
+            agent_key: AgentPubKey,
+            data: &[(DnaFile, Option<MembraneProof>)],
         ) -> ConductorResult<()> {
-            crate::conductor::conductor::genesis_cells(
-                self.clone(),
-                cell_data
-                    .iter()
-                    .map(|(c, p)| (c.as_id().clone(), p.clone()))
-                    .collect(),
+            let payload = crate::sweettest::get_install_app_payload_from_dnas(
+                installed_app_id,
+                agent_key,
+                data,
             )
-            .await?;
-
-            let cell_data = cell_data.into_iter().map(|(c, _)| c);
-            let app = InstalledAppCommon::new_legacy(installed_app_id, cell_data)?;
+            .await;
 
-            // Update the db
-            let _ = self.add_disabled_app_to_db(app).await?;
+            self.install_app_bundle(payload).await?;
```

Comment:
This is the main thrust of this PR: rewriting legacy installation in terms of app bundle installation, to integrate the two code paths. The change to this function signature is what necessitated almost all of the other changes.

Instead of taking an InstalledCell, which specifies the role and agent, it just takes the DNA. This is because we had a lot of code that would allow cells under multiple agents being installed in the same app, and that's not part of our model anymore.